src/main/java/appeng/core/features/AEFeature.java

```diff
@@ -137,6 +156,23 @@ public boolean isVisible()
 	Patterns( Constants.CATEGORY_CRAFTING_FEATURES ),
 	CraftingCPU( Constants.CATEGORY_CRAFTING_FEATURES ),
 
+	BasicCards( Constants.CATEGORY_UPGRADES ),
+	AdvancedCards( Constants.CATEGORY_UPGRADES ),
+	ViewCell( Constants.CATEGORY_UPGRADES ),
+
+	Processors( Constants.CATEGORY_MATERIALS ),
+	PrintedCircuits( Constants.CATEGORY_MATERIALS ),
+	Presses( Constants.CATEGORY_MATERIALS ),
+	CrystalSeeds( Constants.CATEGORY_MATERIALS ),
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00115)